pkg/plugins/app_plugin.go

```diff
@@ -67,6 +72,15 @@ func (app *AppPlugin) Load(decoder *json.Decoder, base *PluginBase, backendPlugi
 		return err
 	}
 
+	if app.Backend {
+		cmd := ComposePluginStartCommand(app.Executable)
+		fullpath := path.Join(app.PluginDir, cmd)
```

Comment:
Need to use `filepath.Join` for filesystem paths.
```suggestion
		fullpath := filepath.Join(app.PluginDir, cmd)
```

Reply:
Does that mean that data source backend plugins aren't working on Windows? 
I just copy-pasted the code from `pkg/plugins/datasource_plugin.go`, where `path` is used.

I can simply replace the import, of course. But I'm a bit concerned now, since I copied these lines from a code that's supposed to be working properly.

What do you think? @aknuds1  Thanks!!!

Reply:
@joanlopez I haven't tested, but the code in pkg/plugins/ should be fixed also to use `filepath.Join` unless I'm missing the obvious.

Reply:
@marefr @papagian Can you confirm whether the code in pkg/plugins/datasource_plugin.go should be fixed to use [filepath.Join](https://golang.org/pkg/path/filepath/#Join) instead of path.Join? So long as it's a filesystem path, we should use filepath.Join, for it to work also on Windows, so unless there's something particular going on here, filepath.Join should be the right choice.

Reply:
Regarding the package documentation (`path` vs `filepath`), you're completely right.

However, there are multiple `path.Join` usages within the `plugins` package. 
Specifically, some of these usages are exactly for the same use case `fullpath := path.Join(r.PluginDir, cmd)`.
So, I guess we can replace all of these, but I'd like to get your confirmation.

Reply:
Regarding how to test it, probably we can add a new method into the `PluginBase` or create a new isolated function (as  `ComposePluginStartCommand`) specifically to get the plugin full path, but it won't prevent future misusages of `path.Join`. 

Thoughts?

Reply:
@aknuds1 @joanlopez I think you're correct, but we haven't had a problem with this before. Therefore I would suggest to handle this in a separate PR to get this PR merged for 7.3 stable and unblock Joan.

Reply:
@marefr I don't think we need to fix the other instances in this PR, but I think my suggestion above should be committed to avoid introducing a new bug.

Reply:
@aknuds1 I do agree! I replaced the `path.Join` usage on this PR.

</br>

> I'm still wondering if we should try to get a test in here.

Also interested in your thoughts 🤔 😇

Reply:
@joanlopez It depends on the deadline, which milestone is it targeting?